compiler/rustc_data_structures/src/sync.rs

```diff
@@ -26,6 +26,18 @@ use std::panic::{catch_unwind, resume_unwind, AssertUnwindSafe};
 pub use std::sync::atomic::Ordering;
 pub use std::sync::atomic::Ordering::SeqCst;
 
+static PARALLEL: std::sync::atomic::AtomicBool = std::sync::atomic::AtomicBool::new(false);
```

Comment:
Follow my earlier suggestion here. This needs to have an uninitialized state for soundness. Changing modes must not be possible once initialized.

Reply:
I think we can use AtomicUsize `0`, `1`, `2` to represent three situations here